bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.juicenet.internal.api;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link JuiceNetApi} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetApi {
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetApi.class);
+
+    private static final String API_HOST = "https://jbv1-api.emotorwerks.com/";
+    private static final String API_ACCOUNT = API_HOST + "box_pin";
+    private static final String API_DEVICE = API_HOST + "box_api_secure";
+
+    protected String apiToken = "";
+    protected JuiceNetHttp httpApi = new JuiceNetHttp();
+    @Nullable
```

Comment:
What's the reason for this to be nullable?

Reply:
Since the class is @NonNullByDefault, it would require bridgeUID to be set in the constructor (there is none for JuiceNetApi). This value is only set in the initialize() function when it is passed as a parameter.

Reply:
Correct, I must have misread something when posting that comment.